astroplan/tests/test_scheduling.py

```diff
@@ -21,70 +22,184 @@
-    assert len(slots) == 2
-    assert slots[0].end == slots[1].start
-
+def test_scheduling_target_down():
+    lco = Observer.at_site('lco')
+    block = [ObservingBlock(FixedTarget.from_name('polaris'), 1 * u.min, 0)]
+    start_time = default_time
+    end_time = start_time + 5*u.day
+    scheduler1 = SequentialScheduler(start_time, end_time, only_at_night, lco,
+                                     transitioner)
+    schedule1 = scheduler1(block)
+    assert len(schedule1.observing_blocks) == 0
+    scheduler2 = PriorityScheduler(start_time, end_time, only_at_night, lco,
+                                   transitioner)
+    schedule2 = scheduler2(block)
+    assert len(schedule2.observing_blocks) == 0
 
-def test_schedule():
-    start_time = Time('2016-02-06 00:00:00')
-    end_time = start_time + 24 * u.hour
-    schedule = Schedule(start_time, end_time)
-    assert schedule.slots[0].start == start_time
-    assert schedule.slots[0].end == end_time
-    assert schedule.slots[0].duration == 24*u.hour
-    schedule.new_slots(0, start_time, end_time)
-    assert len(schedule.slots) == 1
-    new_slots = schedule.new_slots(0, start_time+1*u.hour, start_time+4*u.hour)
-    assert np.abs(new_slots[0].duration - 1*u.hour) < 1*u.second
-    assert np.abs(new_slots[1].duration - 3*u.hour) < 1*u.second
-    assert np.abs(new_slots[2].duration - 20*u.hour) < 1*u.second
 
+def test_scheduling_during_day():
+    block = [ObservingBlock(FixedTarget.from_name('polaris'), 1 * u.min, 0)]
+    day = default_time
+    start_time = apo.midnight(day) + 10*u.hour
+    end_time = start_time + 6*u.hour
+    scheduler1 = SequentialScheduler(start_time, end_time, only_at_night, apo,
+                                     transitioner)
+    schedule1 = scheduler1(block)
+    assert len(schedule1.observing_blocks) == 0
+    scheduler2 = PriorityScheduler(start_time, end_time, only_at_night, apo,
+                                   transitioner)
+    schedule2 = scheduler2(block)
+    assert len(schedule2.observing_blocks) == 0
+# bring this back when MoonIlluminationConstraint is working properly
+'''
+def test_scheduling_moon_up():
+    block = [ObservingBlock(FixedTarget.from_name('polaris'), 1 * u.min, 0)]
+    # on february 23 the moon was up between the start/end times defined below
+    day = default_time + 17 * u.day
+    start_time = apo.midnight(day) - 2 * u.hour
+    end_time = start_time + 6 * u.hour
+    constraints = [AtNightConstraint(), MoonIlluminationConstraint(max=0)]
+    scheduler1 = SequentialScheduler(start_time, end_time, constraints, apo,
+                                     transitioner)
+    schedule1 = scheduler1(block)
+    assert len(schedule1.observing_blocks) == 0
+    scheduler2 = PriorityScheduler(start_time, end_time, constraints, apo,
+                                   transitioner)
+    schedule2 = scheduler2(block)
+    assert len(schedule2.observing_blocks) == 0
+'''
```

Comment:
Yes, #213, without the speedup of the `MoonIlluminationConstraint` it takes really long.